File: web-client/src/app/shared/posts/posts-page.model.ts
import { Post } from './post.model';

export class PostsPage {
    public posts: Post[] = new Array<Post>();
    public curentPageNumber: number = 0;
    public minPageNumber: number = 0;
    public maxPageNumber: number = 0;

    public fromJSON(obj) {
        this.posts = obj.Posts ? obj.Posts.map(p => { let r = new Post(); r.fromJSON(p); return r; }) : obj.Posts;
        this.curentPageNumber = obj.CurentPageNumber;
        this.minPageNumber = obj.MinPageNumber;
        this.maxPageNumber = obj.MaxPageNumber;
    }
}

File: server/Guard/Models/PostsPageModel.cs
using System.Collections.Generic;

namespace Guard.Models
{
    public class PostsPageModel
    {
        public ICollection<PostModel> Posts { get; set; }
        public int CurentPageNumber { get; set; }
        public int MinPageNumber { get; set; }
        public int MaxPageNumber { get; set; }
    }
}
